src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs

```
@@ -194,51 +193,75 @@ public void RemoveAt(int index)
         /// </summary>
         public object SyncRoot => this;
 
-        object IList.this[int index]
+        object? IList.this[int index]
         {
             get
             {
                 return this[index];
             }
             set
             {
-                this[index] = (string)value;
+                if (value is string valueAsString)
```

Before the change if value were not a string, this was an exception, now we handle in gracefully. @dreddy-work - thoughts?

---

while some of the other interface overrides have return value to check on, this one ( couple below) are silently skipping. Given these are public, and documentation isn't very clear on what is expected incase of non string insertions, Maybe we should keep the existing behavior here.

---

I agree, silent skipping opens opportunities for misuse, I would prefer an in-your-face  CastException.

---

Should this behavior be captured using test?